File: 1º-Periodo/IC/C-like/chinelos.c
#include <stdio.h>
#include <stdlib.h>
int main() {
 int n;
 scanf("%d", &n); //numero de tamanhos
 int *estoque = (int *)malloc(n * sizeof(int));
 if (estoque == NULL) {
 printf("Erro ao alocar memória\n");
 return 1;
 } //aloca dinamicamente a quantidade de cada tamanho
 for (int i = 0; i < n; i++) {
 scanf("%d", &estoque[i]);
 }
 int p;
 scanf("%d", &p);
 int cont=0, pedido;
 for (int i = 0; i < p; i++) {
 scanf("%d", &pedido);
 if (estoque[pedido - 1] > 0) {
 estoque[pedido - 1]--;
 cont++;
 } //se o tamanho do pedido for maior que zero (existir estoque) reduz o numero
do estoque e adiciona 1 ao contador
 }
 printf("%d", cont);
 free(estoque);
 return 0;
}

File: 1º-Período/IC/C-like/chinelos.java
import java.util.Scanner;
public class Main {
 public static void main(String[] args) {
 Scanner scanner = new Scanner(System.in);
 int N = scanner.nextInt();
 int[] estoque = new int[N + 1];
 for (int i = 1; i <= N; i++) {
 estoque[i] = scanner.nextInt();
 }
 int P = scanner.nextInt();
 int vendidos = 0;

 for (int i = 0; i < P; i++) {
 int pedido = scanner.nextInt();
 if (pedido >= 1 && pedido <= N && estoque[pedido] > 0) {
 estoque[pedido]--;
 vendidos++;
 } //se o tamanho do pedido for maior que zero (existir estoque) reduz o
numero do estoque e adiciona 1 ao contador
 }
 System.out.println(vendidos);
 scanner.close();
 }
}